core/src/main/scala/akka/persistence/cassandra/query/EventsByPersistenceIdStage.scala

```diff
@@ -550,22 +550,26 @@ import com.github.ghik.silencer.silent
         }
       }
 
-      // The strategy is simple, if full partition was cleaned up we will find null-row
-      // i.e row: <persistence_id>, <partition_nr>, null, ...
-      // If such row exists, we have to switch partition and continue to fetching
-      def checkForGaps(): Unit = {
+      // See PR #509 for background
+      // Only used when gapFreeSequenceNumbers==false
+      // if full partition was cleaned up we look for two empty partitions before completing
+      def checkForGaps(foundEmptyPartitionCount: Int): Unit = {
         session.selectSingleRow(persistenceId, partition).onComplete {
           getAsyncCallback[Try[Option[Row]]] {
             case Success(mbRow) =>
-              val r = mbRow.map(row => (row.getBool("used"), row.getLong("sequence_nr")))
-              r match {
-                case Some((true, _)) =>
-                  //We increment partition nr manually, otherwise we will break `lookingForMissingSeqNr` logic
-                  partition = partition + 1
+              mbRow.map(_.getLong("sequence_nr")) match {
```

Comment:
so we go forward 5 partitions? Seems enough

Reply:
yes, very arbitrary but didn't want to add more config for that "questionable" feature